telegram/bot.py

```diff
@@ -1430,40 +1510,53 @@ def set_webhook(self,
                     max_connections=40,
                     allowed_updates=None,
                     **kwargs):
-        """Use this method to specify a url and receive incoming updates via an outgoing webhook.
+        """
+        Use this method to specify a url and receive incoming updates via an outgoing webhook.
         Whenever there is an update for the bot, we will send an HTTPS POST request to the
-        specified url, containing a JSON-serialized Update. In case of an unsuccessful request, we
-        will give up after a reasonable amount of attempts.
+        specified url, containing a JSON-serialized Update. In case of an unsuccessful request,
+        we will give up after a reasonable amount of attempts.
+
+        If you'd like to make sure that the Webhook request comes from Telegram, we recommend
+        using a secret path in the URL, e.g. https://www.example.com/<token>. Since nobody else
+        knows your bot's token, you can be pretty sure it's us.
 
         Args:
             url (str): HTTPS url to send updates to. Use an empty string to remove webhook
-                integration.
-            certificate (file): Upload your public key certificate so that the root certificate in
-                use can be checked.
+                    integration.
+            certificate (file): Upload your public key certificate so that the root certificate
+                    in use can be checked. See our self-signed guide for details.
             max_connections (Optional[int]): Maximum allowed number of simultaneous HTTPS
-                connections to the webhook for update delivery, 1-100. Defaults to 40. Use lower
-                values to limit the load on your bot's server, and higher values to increase your
-                bot's throughput.
+                    connections to the webhook for update delivery, 1-100. Defaults to 40. Use
+                    lower values to limit the load on your bot's server, and higher values to
+                    increase your bot's throughput.
             allowed_updates (Optional[list[str]]): List the types of updates you want your bot to
-                receive. For example, specify
-                ``["message", "edited_channel_post", "callback_query"]`` to only receive updates of
-                these types. See ``telegram.Update`` for a complete list of available update types.
-                Specify an empty list to receive all updates regardless of type (default). If not
-                specified, the previous setting will be used.
-                Please note that this parameter doesn't affect updates created before the call to
-                the setWebhook, so unwanted updates may be received for a short period of time.
+                    receive. For example, specify ["message", "edited_channel_post",
```

Comment:
We should probably have some constants for these.. Like Message.TYPE.EDITED_CHANNEL_POST or something.